tools/autograd/gen_variable_type.py

```diff
@@ -246,16 +246,15 @@
 PRE_RECORD_TRACE = CodeTemplate("""\
 torch::jit::Node* node = nullptr;
 std::shared_ptr<jit::tracer::TracingState> tracer_state;
-if (jit::tracer::isTracing()) {
-  tracer_state = jit::tracer::getTracingState();
+if (C10_UNLIKELY(jit::tracer::isTracing())) {
+  tracer_state = std::move(jit::tracer::getTracingState());
```

Comment:
This move doesn't do anything `TORCH_API const std::shared_ptr<TracingState>& getTracingState();`, because getTracingState returns a const reference so there still has to be a refcount bump.  To avoid the recount, this can just hold a raw TracingState pointer.

Reply:
Unfortunately, holding a raw pointer doesn't work because `setTracingState` takes a `shared_ptr`.  How bad would it be to change `getTracingState` to return a non-const reference?